lib/+wbs/@Contacts/Contacts.m

```diff
@@ -137,49 +148,36 @@
             % the velocity does not change if there is no impact
             % the velocity change if there is the impact
 
-            % initialize the jacobian relative to the vertices that will be in contact
-            J = [];
-
-            new_contact = false;
-
-            for ii = 1:obj.num_vertices * 2
-                j = (ii - 1) * 3 + 1;
-                % the impact occurs if the point was not in contact and now it is
-                if obj.is_in_contact(ii) == 1 && obj.was_in_contact(ii) == 0
-                    % stack the jacobians of the vertices that NOW are in contact
-                    J = vertcat(J, J_feet(j:j + 2, :));
-                    new_contact = true;
-                end
+            % We shall stack the jacobians relative to the vertices that will be in contact. For
+            % those vertices, the velocity after impact should be zero.
+            mapVerticesNewContact = obj.is_in_contact & ~obj.was_in_contact;
+            new_contact = any(mapVerticesNewContact);
 
-            end
-
-            % if a new contact is detected we should prevent that the velocity of the vertices that previusly were in contact is nonzero.
+            % If a new contact is detected we should make sure that the velocity of the vertices
+            % previously in contact is zero.
             if new_contact
-
-                for ii = 1:obj.num_vertices * 2
-                    j = (ii - 1) * 3 + 1;
-
-                    if obj.was_in_contact(ii)
-                        % stack the jacobian of the vertices that WERE in contact
-                        J = vertcat(J, J_feet(j:j + 2, :));
-                    end
-
-                end
-
+                mapVerticesAtZeroVel = mapVerticesNewContact | obj.was_in_contact;
+            else
+                mapVerticesAtZeroVel = mapVerticesNewContact;
             end
 
-            % compute the projection in the null space of the scaled Jacobian of the vertices if a new contact is detected
-            if ~new_contact
-                return
-            else
+            allIndexes = 1:numel(mapVerticesAtZeroVel);
+            indexesVerticesAtZeroVel = (allIndexes(mapVerticesAtZeroVel)-1)*3+1; % each vertex has 3 components
+            expandedIdxesVerticesAtZeroVel = [indexesVerticesAtZeroVel;indexesVerticesAtZeroVel+1;indexesVerticesAtZeroVel+2];
+            expandedIdxesVerticesAtZeroVel = expandedIdxesVerticesAtZeroVel(:);
+            J = J_feet(expandedIdxesVerticesAtZeroVel,1:end);
```

Comment:
Love your vectorization! :D

Reply:
:D actually it was the easiest solution to have a code supported by the code generator (dynamic matrices building is not supported), and have a variable `J`.